homeassistant/components/onkyo/media_player.py

```diff
@@ -177,130 +190,149 @@ async def async_setup_platform(
-            discovered=discovered,
-        )
-
-        receivers[connection.host] = receiver
-
-        # Discover what zones are available for the receiver by querying the power.
-        # If we get a response for the specific zone, it means it is available.
-        for zone in ZONES:
-            receiver.conn.query_property(zone, "power")
+    all_successful = True
+    for host, result in results:
+        if (
+            result.get("type") == FlowResultType.CREATE_ENTRY
+            or result.get("reason") == "already_configured"
+        ):
+            continue
+        if error := result.get("reason"):
+            all_successful = False
+            async_create_issue(
+                hass,
+                DOMAIN,
+                f"deprecated_yaml_import_issue_{host}_{error}",
+                breaks_in_ha_version="2025.2.0",
+                is_fixable=False,
+                issue_domain=DOMAIN,
+                severity=IssueSeverity.WARNING,
+                translation_key=f"deprecated_yaml_import_issue_{error}",
+                translation_placeholders={
+                    "host": host,
+                    "url": ISSUE_URL_PLACEHOLDER,
+                },
+            )
 
-        # Add the main zone to entities, since it is always active.
-        _LOGGER.debug("Adding Main Zone on %s", receiver.name)
-        main_entity = OnkyoMediaPlayer(
-            receiver, sources, "main", max_volume, receiver_max_volume
+    if all_successful:
+        async_create_issue(
+            hass,
+            HOMEASSISTANT_DOMAIN,
+            f"deprecated_yaml_{DOMAIN}",
+            is_fixable=False,
+            issue_domain=DOMAIN,
+            breaks_in_ha_version="2025.2.0",
+            severity=IssueSeverity.WARNING,
+            translation_key="deprecated_yaml",
+            translation_placeholders={
+                "domain": DOMAIN,
+                "integration_title": "onkyo",
+            },
         )
-        entities["main"] = main_entity
-        async_add_entities([main_entity])
 
-    if host is not None:
-        if host in KNOWN_HOSTS:
-            return
 
-        _LOGGER.debug("Manually creating receiver: %s (%s)", name, host)
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: OnkyoConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up MediaPlayer for config entry."""
+    data = entry.runtime_data
 
-        async def async_onkyo_interview_callback(conn: pyeiscp.Connection) -> None:
-            """Receiver interviewed, connection not yet active."""
-            info = ReceiverInfo(conn.host, conn.port, conn.name, conn.identifier)
-            _LOGGER.debug("Receiver interviewed: %s (%s)", info.model_name, info.host)
-            if info.host not in KNOWN_HOSTS:
-                KNOWN_HOSTS.append(info.host)
-                await async_setup_receiver(info, False, name)
+    receiver = data.receiver
+    all_entities = hass.data[DATA_MP_ENTITIES]
```

Comment:
This dict is created in `async_setup_platform`, I think we want to move it to here

Reply:
Actually it is created in `async_register_services` and it should stay there IMO.